### PR TITLE
Redesign /permissions help

### PR DESCRIPTION
**Pull request by mia-pi-git**

### CODE REVIEW

**Comment by Zarel** [comment]
I personally don't think this looks better. But before we start investigating what we _do_ want, I think we should first split the code apart from the design.

Let's do the implementation first:

```ts
const subPermissions = allPermissions.filter(perm => perm.startsWith('/') && perm.includes(' '));
const subPermissionsByNamespace = {};
for (const perm of subPermissions) {
	const [namespace] = perm.split(' ', 1);
	if (!subPermissionsByNamespace[namespace]) subPermissionsByNamespace[namespace] = [];
	subPermissionsByNamespace[namespace].push(perm);
}
```

**Comment by Zarel** [comment]
After that, the way I'd probably want to render it:

```ts
buffer += `<p><details class="readmore"><summary><strong>Sub-commands:</strong> (will affect one sub-command, like /roomevents view)</summary>`;
for (const subPerms of Object.values(subPermissionsByNamespace) {
	buffer += `<br /><code>` + subPerms.join(`</code> <code>`) + `</code><br />`;
}
buffer += `</details></p>`;
```

**Comment by Zarel** [comment]
What does this look like, btw? Do you have a screenshot and also opinions?

**Comment by mia-pi-git** [comment]
https://prnt.sc/tuwv4n
Personally, i think it's very wide, but you already nixed putting them in their own readmores, so I think this is good.

File: server/chat-commands/room-settings.ts [hunk 1]
```diff
@@ -360,7 +366,11 @@ export const commands: ChatCommands = {
 			buffer += `<p><strong>Single-command permissions:</strong> (will affect one command)<br />`;
 			buffer += `<code>` + permissions.join(`</code> <code>`) + `</code></p>`;
 			buffer += `<p><details class="readmore"><summary><strong>Sub-commands:</strong> (will affect one sub-command, like /roomevents view)</summary>`;
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:368]:
```suggestion
```

I like how you didn't even notice that you repeated this line. :p

Comment (mia-pi-git) [reply server/chat-commands/room-settings.ts:368]: Hey, i noticed when I did it locally :c

Comment (Zarel) [reply server/chat-commands/room-settings.ts:368]: I mean, it's repeated in your screenshot too.

Comment (mia-pi-git) [reply server/chat-commands/room-settings.ts:368]: yes but i thought i'd prioritize screenshot over removing a line

**Comment by Zarel** [comment]
I'm a fan.

**Pull request merged**